File: java/face/src/main/java/co/edu/eafit/MethodFactory.java
package co.edu.eafit;

import co.edu.eafit.interpolation.Vandermonde;
import co.edu.eafit.nonLinearEcuations.Biseccion;
import co.edu.eafit.nonLinearEcuations.IncrementalSearch;
import co.edu.eafit.nonLinearEcuations.Newton;
import co.edu.eafit.nonLinearEcuations.PuntoFijo;
import co.edu.eafit.nonLinearEcuations.RaicesMultiples;
import co.edu.eafit.nonLinearEcuations.ReglaFalsa;
import co.edu.eafit.nonLinearEcuations.Secante;
import co.edu.eafit.systemOfEquations.EliminacionPivoteoParcial;
import co.edu.eafit.systemOfEquations.EliminacionPivoteoTotal;
import co.edu.eafit.systemOfEquations.EliminacionSimple;
import co.edu.eafit.systemOfEquations.FactorizacionCholesky;
import co.edu.eafit.systemOfEquations.FactorizacionCrout;
import co.edu.eafit.systemOfEquations.FactorizacionDoolittle;
import co.edu.eafit.systemOfEquations.FactorizacionGaussiana;
import co.edu.eafit.systemOfEquations.FactorizacionGaussianaPivoteo;
import co.edu.eafit.systemOfEquations.Jacobi;
import co.edu.eafit.systemOfEquations.SOR;
import co.edu.eafit.systemOfEquations.Seidel;

public class MethodFactory {
    public static NumericMethod createMethod(String methodName){
        //Capitulo soluciion de ecuaciones no lineales
    	if (methodName.toLowerCase().equals("newton")) {
            return new Newton();
        }else if (methodName.toLowerCase().equals("busquedas")) {
        	return new IncrementalSearch();
    	}else if(methodName.toLowerCase().equals("biseccion")) {
    		return new Biseccion();
    	}else if(methodName.toLowerCase().equals("regla falsa")) {
    		return new ReglaFalsa();
    	}else if(methodName.toLowerCase().equals("punto fijo")) {
    		return new PuntoFijo();
    	}else if(methodName.toLowerCase().equals("secante")) {
    		return new Secante();
    	}else if(methodName.toLowerCase().equals("raices multiples")) {
    		return new RaicesMultiples();
    	
    	//Capitulo sistemas de ecuaciones
    	}else if(methodName.toLowerCase().equals("eliminacion simple")) {
    		return new EliminacionSimple();
    	}else if(methodName.toLowerCase().equals("eliminacion pivoteo parcial")) {
    		return new EliminacionPivoteoParcial();
    	}else if(methodName.toLowerCase().equals("eliminacion pivoteo total")) {
    		return new EliminacionPivoteoTotal();
    	}else if(methodName.toLowerCase().equals("factorizacion gaussiana")) {
    		return new FactorizacionGaussiana();
    	}else if(methodName.toLowerCase().equals("factorizacion gaussiana pivoteo")) {
    		return new FactorizacionGaussianaPivoteo();
    	}else if(methodName.toLowerCase().equals("crout")) {
    		return new FactorizacionCrout();
    	}else if(methodName.toLowerCase().equals("doolittle")) {
    		return new FactorizacionDoolittle();
    	}else if(methodName.toLowerCase().equals("cholesky")) {
    		return new FactorizacionCholesky();
    	}else if(methodName.toLowerCase().equals("jacobi")) {
    		return new Jacobi();
    	}else if(methodName.toLowerCase().equals("seidel")) {
    		return new Seidel();
    	}else if(methodName.toLowerCase().equals("sor")) {
    		return new SOR();
    	
    	// Capitulo interpolacion
    	}else if(methodName.toLowerCase().equals("vandermonde")) {
    		return new Vandermonde();
    	}
    	else{
            return new NotImplementedMethod();
        }
    }
}

File: python/face/api/metodos/method_factory.py
from .numeric_method import NumericMethod

from .ecuaciones_no_lineales.metodo_biseccion import Bisection
from .ecuaciones_no_lineales.metodo_busquedas import IncrementalSearch
from .ecuaciones_no_lineales.metodo_regla_falsa import FalsePosition
from .ecuaciones_no_lineales.metodo_punto_fijo import FixedPoint
from .ecuaciones_no_lineales.metodo_newton import NewtonMethod
from .ecuaciones_no_lineales.metodo_newton_m import Newton2Method
from .ecuaciones_no_lineales.metodo_secante import Secante

from .sistemas_ecuaciones.eliminacion_simple import EliminacionSimple
from .sistemas_ecuaciones.eliminacion_p_parcial import EliminacionPivoteoParcial
from .sistemas_ecuaciones.eliminacion_p_total import EliminacionPivoteoTotal
from .sistemas_ecuaciones.factorizacion_gaussiana import FactorizacionGaussiana
from .sistemas_ecuaciones.factorizacion_pivoteo import FactorizacionPivoteo
from .sistemas_ecuaciones.crout import FactorizacionCrout
from .sistemas_ecuaciones.doolittle import FactorizacionDoolittle
from .sistemas_ecuaciones.cholesky import FactorizacionCholesky
from .sistemas_ecuaciones.jacobi import Jacobi
from .sistemas_ecuaciones.seidel import Seidel
from .sistemas_ecuaciones.sor import SOR

from .interpolacion.sistema_ecuaciones import MetodoSistemaEcuaciones
from .interpolacion.newton_diferencias import NewtonDiferenciasDivididas
from .interpolacion.spline_lineal import SplinesLineales


def create_method(method):
    method = method.lower()
    # Capitulo ecuaciones no lineales
    if method == "busquedas":
        return IncrementalSearch()
    elif method == "biseccion":
        return Bisection()
    elif method == "regla_falsa":
        return FalsePosition()
    elif method == "punto_fijo":
        return FixedPoint()
    elif method == "newton":
        return NewtonMethod()
    elif method == "newton2":
        return Newton2Method()
    elif method == "secante":
        return Secante()

    # Capitulo sistemas de ecuaciones
    elif method == "eliminacion_simple":
        return EliminacionSimple()
    elif method == "eliminacion_piv_parcial":
        return EliminacionPivoteoParcial()
    elif method == "eliminacion_piv_total":
        return EliminacionPivoteoTotal()

    elif method == "factorizacion_simple":
        return FactorizacionGaussiana()
    elif method == "factorizacion_pivoteo":
        return FactorizacionPivoteo()
    elif method == "doolittle":
        return FactorizacionDoolittle()
    elif method == "crout":
        return FactorizacionCrout()
    elif method == "cholesky":
        return FactorizacionCholesky()
    elif method == "jacobi":
        return Jacobi()
    elif method == "seidel":
        return Seidel()
    elif method == "sor":
        return SOR()

    # Capitulo de interpolacion
    elif method == "interpolacion_ecuaciones":
        return MetodoSistemaEcuaciones()
    elif method == "newton_diferencias":
        return NewtonDiferenciasDivididas()
    elif method == "spline_lineal":
        return SplinesLineales()
    else:
        return NumericMethod()
